File: packages/caption-schema/src/job.ts
import { z } from "zod";

export const JobTypeSchema = z.enum([
  "MEDIA_ANALYSIS",
  "AUDIO_EXTRACTION",
  "TRANSCRIPTION",
  "SUBTITLE_IMPORT",
  "CAPTION_GENERATION",
]);

export type JobType = z.infer<typeof JobTypeSchema>;

export const JobStatusSchema = z.enum([
  "QUEUED",
  "PROCESSING",
  "COMPLETED",
  "FAILED",
  "CANCELLED",
]);

export type JobStatus = z.infer<typeof JobStatusSchema>;

export const JobProgressEventSchema = z.object({
  jobId: z.string(),
  status: JobStatusSchema,
  stage: z.string(),
  progress: z.number().min(0).max(100),
  message: z.string(),
  error: z.string().nullable().optional(),
  createdAt: z.string().optional(),
  completedAt: z.string().nullable().optional(),
});

export type JobProgressEvent = z.infer<typeof JobProgressEventSchema>;

File: apps/api/app/schemas/job.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class JobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    job_type: str
    status: str
    stage: str
    progress: float
    message: str
    error_details: Optional[str] = None
    created_at: Optional[datetime] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None


class JobProgressEvent(BaseModel):
    job_id: str
    project_id: str
    status: str
    stage: str
    progress: float
    message: str
    error: Optional[str] = None
